homeassistant/components/snoo/coordinator.py

```diff
@@ -0,0 +1,39 @@
+"""Support for Anova Coordinators."""
+
+import logging
+
+from python_snoo.containers import SnooData, SnooDevice
+from python_snoo.snoo import Snoo
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class SnooCoordinator(DataUpdateCoordinator[SnooData]):
+    """Snoo custom coordinator."""
+
+    config_entry: ConfigEntry
```

Comment:
Let's use the typed config entry here

Reply:
Since the typed config entry has the coordinator on it, it becomes a circular dependency.

Is best practice in HA to move the typed config entry to a models.py or common.py - Import coordinator with if typing.TYPE_CHECKING

And then import the typed config entry from the models.py elsewhere?

Reply:
We usually have the type in the coordinator.py now